### ISSUE TITLE
Be able to click without leaving grid mode.

### ISSUE THREAD

**Issue by alanxoc3** [rvaiya/warpd#248]
I don't see much of a reason for normal mode, because you can move around in gridmode just as easily (grid_left/right/up/down). So I'd like to be able to stay in grid mode when I click/drag/scroll.

When I click, I get sent back to normal mode. Is it possible to not get sent out of grid mode when clicking? If not, is this something that could be supported?

**Comment by rvaiya** [comment]
Grid mode exists as an alternative to hint mode for macroscopic movements, with the idea being that once the user has selected a target, they are free to choose another mode for the next macroscopic movement using one of the modal triggers. 
Once the grid has been narrowed to a target it is mostly useless, and the user will probably want to reactivate it. Having said that, the recommended way to use warpd is to use hint mode for macroscopic movements and normal mode for more refined ones.

**Comment by alanxoc3** [comment]
Yeah, I understand that workflow and this is a very minor request. I also agree that grid mode is usually not very useful after you've narrowed down to a specific place, but here is the specific use case for me...

I want to do this: I start warpd in grid mode, then I scroll through a webpage (which is full screen & my cursor is in the center of the screen), then I narrow the grid down to a specific place.

But I'm forced to do this: I start warpd in grid mode, then I scroll through a webpage (which takes me out of grid mode), then I enter grid mode again, then I narrow the grid down to a specific place.

**Comment by rvaiya** [comment]
If I understand you correctly, this only saves a single keystroke. Grid mode can be entered from normal mode by pressing `g`, so you can just activate warpd in normal mode, scroll (`e`/`r`) and press `g` to initiate a grid movement. Scrolling while in grid mode also has the disadvantage of obscuring the screen.

**Comment by alanxoc3** [comment]
You understand correctly, it only saves 1 keystroke for me and is totally not a big deal.

Just playing with the idea, exposing mouse keys to modes outside of normal mode might make these scenarios possible too:
- Start in hint mode, start a drag but don't leave hint mode, jump to a different hint to extend the drag, end the drag. Saves 1 key press from entering hint mode again.
- Start in grid mode, start a drag and starting the drag resets the grid, jump to different grid location, end the drag. Saves 1 key press from entering grid mode again.

**Comment by sathishmanohar** [comment]
I use warpd in mostly hint mode. I've wanted something similar as well. My use case is I want to quickly close the a few tray icons. So multiple times I have to do

- set cursor to the right location
- right click
- click on quit
- repeat

For this I currently have to leave and exit warpd multiple times but I am able to do the clicking in the normal mode. I'd be able to click and do whatever I want and then invoke the hint again with x.
